internal/store/postgres/resource_repository.go

```diff
@@ -213,3 +214,46 @@ func (r ResourceRepository) GetByURN(ctx context.Context, urn string) (resource.
 
 	return resourceModel.transformToResource(), nil
 }
+
+func buildGetResourcesByNamespaceQuery(dialect goqu.DialectWrapper, name string, ns namespace.Namespace) (string, interface{}, error) {
+	namespaceQueryExpression := goqu.Ex{
+		"backend": goqu.L(ns.Backend),
+	}
+
+	if ns.ResourceType != "" {
+		namespaceQueryExpression["resource_type"] = goqu.L(ns.ResourceType)
+	}
+
+	getNamespaceQuery := dialect.Select("id").From(TABLE_NAMESPACES).Where(namespaceQueryExpression)
+	getResourcesByURNQuery, params, err := dialect.Select(&ResourceCols{}).From(TABLE_RESOURCES).Where(goqu.Ex{
+		"name":         goqu.L(name),
+		"namespace_id": goqu.Op{"in": getNamespaceQuery},
+	}).ToSQL()
+
+	return getResourcesByURNQuery, params, err
+}
+
+func (r ResourceRepository) GetByNamespace(ctx context.Context, name string, ns namespace.Namespace) (resource.Resource, error) {
+	var fetchedResource Resource
+
+	query, params, err := buildGetResourcesByNamespaceQuery(dialect, name, ns)
+	if err != nil {
+		return resource.Resource{}, fmt.Errorf("%w: %s", queryErr, err)
+	}
+
+	err = r.dbc.WithTimeout(ctx, func(ctx context.Context) error {
+		return r.dbc.GetContext(ctx, &fetchedResource, query, params)
+	})
+
+	if errors.Is(err, sql.ErrNoRows) {
+		return resource.Resource{}, resource.ErrNotExist
+	} else if err != nil {
+		return resource.Resource{}, fmt.Errorf("%w: %s", dbErr, err)
+	}
+
+	if err != nil {
+		return resource.Resource{}, err
+	}
```

Comment:
CMMIW, if `err != nil`, this always returned in line `251` right?
So what is the point of adding line `254-256`?

Reply:
+1. this `if err != nil` block is unreachable.

Reply:
also, it is preferrable to indent error flows. 

```golang
err = r.dbc.WithTimeout(....)
if err != nil {
    if errors.Is(err, sql.ErrNoRows) {
        return resource.Resource{}, resource.ErrNotExist
    }
    return resource.Resource{}, fmt.Errorf("%w: %s", dbErr, err)
}
```

The reason is it makes it easier to avoid the kind of mistakes that's been made here..